File: frontend/src/api/architecture.ts
import { apiClient } from "@/lib/api";
import type { ArchitectureReport } from "@/types/api";

export const ArchitectureApi = {
  report: async (repositoryId: string): Promise<ArchitectureReport> => {
    const { data } = await apiClient.get<ArchitectureReport>(
      `/repositories/${repositoryId}/architecture`,
    );
    return data;
  },
};

File: backend/app/api/v1/endpoints/architecture.py
"""Architecture-discovery endpoint."""
from __future__ import annotations

import uuid

from fastapi import APIRouter, Depends

from app.core.dependencies import ArchitectureServiceDep, verify_repository_access
from app.schemas.architecture import ArchitectureReport

router = APIRouter(
    prefix="/repositories/{repository_id}",
    tags=["architecture"],
    dependencies=[Depends(verify_repository_access)],
)


@router.get(
    "/architecture",
    response_model=ArchitectureReport,
    summary="Discover layers, components, and produce a Mermaid diagram",
)
async def get_architecture(
    repository_id: uuid.UUID,
    service: ArchitectureServiceDep,
) -> ArchitectureReport:
    return await service.report(repository_id)
